pkg/secrets/keyvault/client.go

```diff
@@ -307,55 +307,112 @@ func (k *KeyvaultSecretClient) Delete(ctx context.Context, key types.NamespacedN
 	return err
 }
 
-// Get gets a key from KeyVault
-func (k *KeyvaultSecretClient) Get(ctx context.Context, key types.NamespacedName) (map[string][]byte, error) {
-	vaultBaseURL := getVaultsURL(ctx, k.KeyVaultName)
-	var secretName string
-	if len(key.Namespace) != 0 {
-		secretName = key.Namespace + "-" + key.Name
+// Delete deletes a key in KeyVault
+func (k *SecretClient) Delete(ctx context.Context, key secrets.SecretKey, opts ...secrets.SecretOption) error {
+	options := &secrets.Options{}
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	if options.Flatten {
+		for _, suffix := range options.FlattenSuffixes {
+			secretName, err := k.makeSecretName(secrets.SecretKey{Name: key.Name + "-" + suffix, Namespace: key.Namespace, Kind: key.Kind})
+			if err != nil {
+				return errors.Wrapf(err, "couldn't make secret name for %+v", key)
+			}
+			err = k.deleteKeyVaultSecret(ctx, secretName)
+			if err != nil {
+				return err
+			}
+		}
+		return nil
 	} else {
-		secretName = key.Name
+		secretName, err := k.makeSecretName(key)
+		if err != nil {
+			return errors.Wrapf(err, "couldn't make secret name for %+v", key)
```

Comment:
Don't we run a PII disclosure risk using `%+v` here with a secret?

Reply:
`key` is just the name of the secret, so `+v` is safe.

With that said maybe I should just override `Stringer` and use that for the key.... that probably is safer.

Reply:
Fixed this and also improved error message of `makeSecretName` so wrapping was not needed here (the error from `makeSecretName` now includes the requisite details via the `Stringer` interface).